store/sqlstore/channel_store.go

```diff
@@ -3428,29 +3443,59 @@ func (s SqlChannelStore) SearchGroupChannels(userId, term string) (model.Channel
 	return groupChannels, nil
 }
 
-// TODO: rewrite in squirrel (https://github.com/mattermost/mattermost-server/issues/19336)
-func (s SqlChannelStore) GetMembersByIds(channelId string, userIds []string) (model.ChannelMembers, error) {
-	var dbMembers channelMemberWithSchemeRolesList
+func (s SqlChannelStore) GetMembersByIds(channelID string, userIDs []string) (model.ChannelMembers, error) {
+	if len(userIDs) == 0 {
+		return nil, errors.New("no user IDs were provided")
+	}
```

Comment:
This wasn't there previously. Let's remove it to keep the behavior same as earlier. In general, all input validations should be done in the app layer.

Reply:
This is here to make the method conform to previous behavior. GORP seems to error when an argument is an empty list. One of the tests (I don't remember if for `GetMembersByIds` or `GetMembersByChannelIds`, I can look it up for you) explicitly checks that the method errors when provided an empty list.

Reply:
I am wondering if this is due to the nil slice we are passing instead of the empty slice. Can you make the change to empty slice and see if you still get an error? If we do, we might need to change the returned error to pass the test.

Reply:
The problem is this test (storetest/channel_store.go:6324):
```go
_, nErr = ss.Channel().GetMembersByIds(m1.ChannelId, []string{})
require.Error(t, nErr, "empty user ids - should have failed")
```
it expects `GetMembersByIds` to return an error when userIDs is empty. I could simply remove this test but I suspects it's there for a reason.

Reply:
Oh dear, the test was actually testing the wrong thing. The old code did 

```go
if _, err := s.GetReplica().Select(&dbMembers, channelMembersForTeamWithSchemeSelectQuery+"WHERE ChannelMembers.ChannelId = :ChannelId AND ChannelMembers.UserId IN "+keys, 
```

which means that when keys was empty, then it would essentially be an incorrect SQL statement, and indeed, when I print the error, that's what I see :)

```
=== RUN   TestChannelStore/MySQL/GetMembersByIds
    channel_store.go:6325: failed to find ChannelMembers with channelId=64zgnfped3rr8c8kjc3epe9yqo and userId in []: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')' at line 19
=== RUN   TestChannelStore/PostgreSQL
=== RUN   TestChannelStore/PostgreSQL/GetMembersByIds
    channel_store.go:6325: failed to find ChannelMembers with channelId=ogi3kwf1zpymzms61qk6wedcda and userId in []: pq: syntax error at or near ")"
```

So we are good to just remove the test. Please apply this patch:

```diff
diff --git a/store/sqlstore/channel_store.go b/store/sqlstore/channel_store.go
index 6346e43ed..72e4c3c4e 100644
--- a/store/sqlstore/channel_store.go
+++ b/store/sqlstore/channel_store.go
@@ -3509,10 +3509,6 @@ func (s SqlChannelStore) SearchGroupChannels(userId, term string) (model.Channel
 }

 func (s SqlChannelStore) GetMembersByIds(channelID string, userIDs []string) (model.ChannelMembers, error) {
-       if len(userIDs) == 0 {
-               return nil, errors.New("no user IDs were provided")
-       }
-
        query := s.channelMembersForTeamWithSchemeSelectQuery.Where(
                sq.Eq{
                        "ChannelMembers.ChannelId": channelID,
@@ -3526,7 +3522,6 @@ func (s SqlChannelStore) GetMembersByIds(channelID string, userIDs []string) (mo
        }

        dbMembers := channelMemberWithSchemeRolesList{}
-
        if err := s.GetReplicaX().Select(&dbMembers, sql, args...); err != nil {
                return nil, errors.Wrapf(err, "failed to find ChannelMembers with channelId=%s and userId in %v", channelID, userIDs)
        }
diff --git a/store/storetest/channel_store.go b/store/storetest/channel_store.go
index 1449ce739..3e7343af7 100644
--- a/store/storetest/channel_store.go
+++ b/store/storetest/channel_store.go
@@ -6321,8 +6321,9 @@ func testChannelStoreGetMembersByIds(t *testing.T, ss store.Store) {
        require.NoError(t, nErr, nErr)
        require.Len(t, members, 2, "return wrong number of results")

-       _, nErr = ss.Channel().GetMembersByIds(m1.ChannelId, []string{})
-       require.Error(t, nErr, "empty user ids - should have failed")
+       members, nErr = ss.Channel().GetMembersByIds(m1.ChannelId, []string{})
+       require.NoError(t, nErr)
+       require.Len(t, members, 0)
 }
```

Reply:
I've applied the patch and the tests pass :)